lms/lib/courseware_search/lms_filter_generator.py

```diff
@@ -3,13 +3,35 @@
     * Filter by all courses in which the user is enrolled in
 """
 from student.models import CourseEnrollment
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
 
 from search.filter_generator import SearchFilterGenerator
+from openedx.core.djangoapps.course_groups.cohorts import get_cohort
+from openedx.core.djangoapps.course_groups.models import CourseUserGroupPartitionGroup
 
 
 class LmsSearchFilterGenerator(SearchFilterGenerator):
     """ SearchFilterGenerator for LMS Search """
 
+    def filter_dictionary(self, **kwargs):
+        """ base implementation which filters via start_date """
+        filter_dictionary = super(LmsSearchFilterGenerator, self).filter_dictionary(**kwargs)
+        if 'user' in kwargs and 'course_id' in kwargs and kwargs['course_id']:
+            try:
+                course_key = CourseKey.from_string(kwargs['course_id'])
+            except InvalidKeyError:
+                course_key = SlashSeparatedCourseKey.from_deprecated_string(kwargs['course_id'])
+            cohort = get_cohort(kwargs['user'], course_key, assign=False)
```

Comment:
This code directly bypasses the cohort scheme object, and hence doesn't obey the preview masquerade logic. Instead, you should mimic the logic in ``lms/djangoapps/courseware/access.py``. That is looking up the group for each partition, but in your case you only need to do it for the one. Your code should be something like:

    cohorted_user_partition = get_cohorted_user_partition(course_key)
    partition_group = partition.scheme.get_group_for_user(
            course_key,
            user,
            cohorted_user_partition,
        )

Let me know if this doesn't work or if you have any questions.